root/static/scripts/common/components/EntityLink.js

```diff
@@ -32,96 +36,109 @@ const DeletedLink = ({name, allowNew}) => {
-  render() {
-    let area = this.props.area;
-
-    if (!area.ended) {
-      return null;
-    }
+  return (
+    <Frag>
+      {dates ? bracketed(dates) : null}
+      {event.cancelled
+        ? <Comment className="cancelled" comment={l('cancelled')} />
+        : null}
+    </Frag>
+  );
+};
 
-    let comment;
-    let beginYear = area.begin_date ? area.begin_date.year : null;
-    let endYear = area.end_date ? area.end_date.year : null;
+const AreaDisambiguation = ({area}: {|+area: AreaT|}) => {
+  if (!area.ended) {
+    return null;
+  }
 
-    if (beginYear && endYear) {
-      comment = l('historical, {begin}-{end}', {begin: beginYear, end: endYear});
-    } else if (endYear) {
-      comment = l('historical, until {end}', {end: endYear});
-    } else {
-      comment = l('historical');
-    }
+  let comment;
+  const beginYear = area.begin_date ? area.begin_date.year : null;
+  const endYear = area.end_date ? area.end_date.year : null;
 
-    return <Comment className="historical" comment={comment} />;
+  if (beginYear && endYear) {
+    comment = l('historical, {begin}-{end}', {begin: beginYear, end: endYear});
+  } else if (endYear) {
+    comment = l('historical, until {end}', {end: endYear});
+  } else {
+    comment = l('historical');
   }
-}
+
+  return <Comment className="historical" comment={comment} />;
+};
 
 const NoInfoURL = ({url, allowNew}) => (
   <Frag>
     <a href={url}>{url}</a>
     {' '}
-    <DeletedLink name={'[' + l('info') + ']'} allowNew={allowNew} />
+    <DeletedLink allowNew={allowNew} name={'[' + l('info') + ']'} />
   </Frag>
 );
 
-const EntityLink = (props = {}) => {
-  let {
-    allowNew,
-    content,
-    entity,
-    hover,
-    showDeleted = true,
-    showDisambiguation,
-    subPath,
-    ...anchorProps
-  } = props;
-
+/* eslint-disable sort-keys, flowtype/sort-keys */
+type EntityLinkProps = {
+  +allowNew?: boolean,
+  +content?: React.Node,
+  +entity: CoreEntityT,
+  +hover?: string,
+  +showDeleted?: boolean,
+  +showDisambiguation?: boolean,
+  +subPath?: string,
+
+  // ...anchorProps
+  href?: string,
+  title?: string,
+  +target?: '_blank',
+};
+/* eslint-enable sort-keys, flowtype/sort-keys */
+
+const EntityLink = ({
+  allowNew = false,
+  content,
+  entity,
+  hover,
+  showDeleted = true,
+  showDisambiguation,
+  subPath,
+  ...anchorProps
+}: EntityLinkProps) => {
   const hasCustomContent = nonEmpty(content);
-  const entityType = entity.entityType;
-  const comment = ko.unwrap(entity.comment);
+  const comment = entity.comment ? ko.unwrap(entity.comment) : '';
 
   if (showDisambiguation === undefined) {
     showDisambiguation = !hasCustomContent;
   }
 
-  if (entityType === 'artist' && !nonEmpty(hover)) {
+  if (entity.entityType === 'artist' && !nonEmpty(hover)) {
     hover = entity.sort_name + bracketed(comment);
   }
 
-  if (entityType === 'artist' || entityType === 'instrument') {
-    content = content || entity.l_name;
+  if (entity.entityType === 'area') {
```

Comment:
See 75252bf82a090f255f1ad3fb7bbbf36add5302ad. I only discovered this bug because of the added types (Flow complained about `l_name` not existing, which caused me to dig deeper and see that the condition was also wrong). So they're already helping. :)